packages/@lwc/synthetic-shadow/src/faux-shadow/shadow-root.ts

```diff
@@ -627,6 +628,18 @@ export function SyntheticShadowRoot() {
 }
 SyntheticShadowRoot.prototype = create(DocumentFragment.prototype, SyntheticShadowRootDescriptors);
 
+// `this.shadowRoot instanceof ShadowRoot` should evaluate to true even for synthetic shadow
+defineProperty(SyntheticShadowRoot, Symbol.hasInstance, {
+    value: function (object: any): boolean {
+        return (
+            !isUndefined(object) &&
+            !isNull(object) &&
+            (isInstanceOfNativeShadowRoot(object) ||
+                getPrototypeOf(object) === SyntheticShadowRoot.prototype)
```

Comment:
It would be "more accurate" to climb up the protochain instead of just checking the proto if you want to stick to the spec. I guess, it a completely reasonable assumption to consider that a shadow root is never subclassed.

Reply:
I'll add a comment to this effect. Yes, I'm not bothering to check sub-prototypes.